pkg/operator/controller/ingress/controller.go

```diff
@@ -412,7 +417,8 @@ var (
 		configv1.VersionTLS10: {},
 		configv1.VersionTLS11: {},
 		configv1.VersionTLS12: {},
-		configv1.VersionTLS13: {},
+		// TODO: Add VersionTLS13 support after haproxy is built with an openssl
+		//  version that supports tls v1.3.
```

Comment:
If the TLS profile specifies the "Custom" profile type with TLS 1.3 as the minimum TLS version, the ingresscontroller will be rejected.  This could happen, for example, with the default ingresscontroller if the APIServer config specifies such a profile.  We do have a pretty severe warning of the risk of catastrophe from using the "Custom" profile type, so I am not saying that we should not reject an ingresscontroller under these circumstances, but I want to make sure everybody understands and that we have a consensus.  @ironcladlou

Reply:
Sounds good to me given the caveats we're documenting for custom, what does @deads2k think?

Reply:
I think failing when there is an impossible custom profile is best.  If a custom profile lists tls 1.2 and 1.3, I expect the custom profile to work, but only with serving 1.2 (filtering similar to how we handle others).  If a custom profile only lists 1.3 and it is filtered out, I would expect to see a failure applying the new profile.

The overall change to rejiggle modern into intermediate seems pretty questionable.  We do reserve the right to decide what these values means, but why is it so hard to update the level of openssl?

Reply:
> I think failing when there is an impossible custom profile is best.

@Miciah I believe the PR in its current state meets this requirement. A custom profile that specifies `minTLSVersion: VersionTLSv13` is impossible and would not be admitted.

> If a custom profile lists tls 1.2 and 1.3, I expect the custom profile to work, but only with serving 1.2 (filtering similar to how we handle others). If a custom profile only lists 1.3 and it is filtered out, I would expect to see a failure applying the new profile.

@deads2k the `TLSSecurityProfile` API only specifies [MinTLSVersion](https://github.com/openshift/api/blob/master/config/v1/types_tlssecurityprofile.go#L160-L166). `MaxTLSVersion` was dropped from the api at the final stages. Since the current haproxy template router does not support TLS v1.3, `VersionTLS13` is impossible.

> why is it so hard to update the level of openssl

@frobware is working on upgrading haproxy, so I'll let him comment on the details.

Reply:
> > I think failing when there is an impossible custom profile is best.
> 
> @Miciah I believe the PR in its current state meets this requirement. A custom profile that specifies `minTLSVersion: VersionTLSv13` is impossible and would not be admitted.

This PR will indeed reject a custom profile that specifies TLS 1.3.  One possible improvement would be also to filter out TLS 1.3 ciphers when validating a custom profile.  `validateTLSSecurityProfile` already rejects a custom profile with an empty ciphers list, so we could add the filtering before the existing emptiness check.

Reply:
> > why is it so hard to update the level of openssl
> 
> @frobware is working on upgrading haproxy, so I'll let him comment on the details.

We don't necessarily need to upgrade HAProxy; it should suffice to compile the current HAProxy version against OpenSSL 1.1.1.  The challenge is that getting OpenSSL 1.1.1 would require building on a RHEL8 base image.